src/app/select/select.directive.ts

```diff
@@ -205,4 +215,27 @@ export class MzSelectDirective extends HandlePropChanges implements AfterViewIni
 
     this.renderer.invokeElementMethod(this.selectElement, 'material_select');
   }
+
+  listenOptionChanges() {
+    Observable.fromEvent($(this.selectElement), 'DOMSubtreeModified')
+      .debounceTime(100)
+      .filter((event: JQueryEventObject) => {
+        const currentOptions: any = Array.from(event.currentTarget.children);
```

Comment:
You can remove cast to `any` by replacing this line with `const currentOptions = event.currentTarget.children;` and use `Array.from(currentOptions).some(...)` at Line 232